lib/betagouvbot/mail.rb

```diff
@@ -5,42 +5,65 @@
 
 module BetaGouvBot
   class Mail
-    attr_accessor :subject
-    attr_accessor :recipients
+    class << self
+      def rules
+        @rules ||= {
+          21 => { mail: from_file('data/mail_3w.md', ['{{author.id}}@beta.gouv.fr']) },
+          14 => { mail: from_file(
+            'data/mail_2w.md',
+            ['{{author.id}}@beta.gouv.fr', 'contact@beta.gouv.fr']
+          ) },
+          1  => { mail: from_file('data/mail_1day.md', ['{{author.id}}@beta.gouv.fr']) },
+          -1 => { mail: from_file('data/mail_after.md', ['contact@beta.gouv.fr']) }
+        }
+      end
 
-    def self.from_file(body_path, recipients = [], sender = 'secretariat@beta.gouv.fr')
-      # Email data files consist of 1 subject line plus body
-      subject, *rest = File.readlines(body_path)
-      Mail.new(subject.strip, rest.join, recipients, sender)
+      # @note Email data files consist of 1 subject line plus body
+      def from_file(body_path, recipients = [], sender = 'secretariat@beta.gouv.fr')
+        subject, *rest = File.readlines(body_path)
+        new(subject.strip, rest.join, recipients, sender)
+      end
     end
 
+    attr_reader :subject, :body_t, :recipients, :sender
+
     def initialize(subject, body_t, recipients, sender)
-      @subject = subject
-      @body_t = body_t
+      @subject    = subject
+      @body_t     = body_t
       @recipients = recipients
-      @sender = sender
+      @sender     = sender
     end
 
     def format(context)
-      md_source = self.class.render(@body_t, context)
+      md_source = render_template(body_t, context)
       { 'personalizations': [{
-        'to': @recipients.map { |mail| { 'email' => self.class.render(mail, context) } },
-        'subject': self.class.render(@subject, context)
+        'to': recipients.map { |mail| { 'email' => render_template(mail, context) } },
+        'subject': render_template(subject, context)
       }],
-        'from': { 'email' => self.class.render(@sender, context) },
+        'from': { 'email' => render_template(sender, context) },
         'content': [{
           'type': 'text/html',
-          'value': Kramdown::Document.new(md_source).to_html
+          'value': render_document(md_source)
         }] }
     end
 
-    def self.template_factory
+    private
+
+    def render_template(template, context)
```

Comment:
There's just one change here : neither ```.template_factory``` nor ```.render``` where _actual_ class methods, as they were never called from an external client. The only thing I did was to declare them correctly the way the class itself uses them : private, instance methods.

I understand you want to pair on this, but I'm having a hard time justifying the investment for something that seems pretty straightforward and standard practice.

I'd rather prefer to toss those changes away, but that's complicated also : does it mean we'll have to get into a pair IRL or toss away logic for all contributions?